mono/mini/interp/transform.c

```diff
@@ -6226,14 +6241,16 @@ interp_cprop (TransformData *td)
 	StackContentInfo *sp = stack;
 	InterpInst *ins;
 	int last_il_offset = -1;
+	gboolean const inlining = td->inlined_method != NULL;
 
 	for (ins = td->first_ins; ins != NULL; ins = ins->next) {
 		int pop, push;
 		int il_offset = ins->il_offset;
 		// Optimizations take place only inside a single basic block
 		// If two instructions have the same il_offset, then the second one
 		// cannot be part the start of a basic block.
-		gboolean is_bb_start = il_offset != -1 && td->is_bb_start [il_offset] && il_offset != last_il_offset;
+		// Only single basic blocks are inlined.
+		gboolean is_bb_start = il_offset != -1 && (inlining || td->is_bb_start [il_offset]) && il_offset != last_il_offset;
```

Comment:
We don't need this. Optimizations happen once all code has been generated. The concept of inlining no longer exists. Also il_offsets are only for the original method (if we inline a call, all instructions generated from the inlined method will have the same il_offset as the call instruction from the parent)